styx-scheduler-service/src/test/java/com/spotify/styx/workflow/ParameterUtilTest.java

```diff
@@ -114,4 +117,72 @@ public void shouldTruncateInstant() throws Exception {
     final Instant months = truncateInstant(TIME, Partitioning.MONTHS);
     assertThat(months, is(truncatedTimeMonths));
   }
+
+  @Test
+  public void shouldRangeOfInstantsHours() throws Exception {
+    final Instant startInstant = Instant.parse("2016-12-31T23:00:00.00Z");
+    final Instant endInstant = Instant.parse("2017-01-01T01:00:00.01Z");
+
+    List<Instant> list = rangeOfInstants(startInstant, endInstant, Partitioning.HOURS);
+    assertThat(list, contains(
+        Instant.parse("2016-12-31T23:00:00.00Z"),
+        Instant.parse("2017-01-01T00:00:00.00Z"),
+        Instant.parse("2017-01-01T01:00:00.00Z"))
+    );
+  }
+
+  @Test
+  public void shouldRangeOfInstantsDays() throws Exception {
+    final Instant startInstant = Instant.parse("2016-12-31T00:00:00.00Z");
+    final Instant endInstant = Instant.parse("2017-01-02T00:00:00.01Z");
+
+    List<Instant> list = rangeOfInstants(startInstant, endInstant, Partitioning.DAYS);
+    assertThat(list, contains(
+        Instant.parse("2016-12-31T00:00:00.00Z"),
+        Instant.parse("2017-01-01T00:00:00.00Z"),
+        Instant.parse("2017-01-02T00:00:00.00Z"))
+    );
+  }
+
+  @Test
+  public void shouldRangeOfInstantsWeeks() throws Exception {
+    final Instant startInstant = Instant.parse("2016-12-30T00:00:00.00Z");
+    final Instant endInstant = Instant.parse("2017-01-13T00:00:00.01Z");
+
+    List<Instant> list = rangeOfInstants(startInstant, endInstant, Partitioning.WEEKS);
+    assertThat(list, contains(
+        Instant.parse("2016-12-30T00:00:00.00Z"),
+        Instant.parse("2017-01-06T00:00:00.00Z"),
+        Instant.parse("2017-01-13T00:00:00.00Z"))
+    );
+  }
+
+  @Test
+  public void shouldRangeOfInstantsMonths() throws Exception {
+    final Instant startInstant = Instant.parse("2017-01-31T00:00:00.00Z");
+    final Instant endInstant = Instant.parse("2017-03-28T00:00:00.01Z");
+
+    List<Instant> list = rangeOfInstants(startInstant, endInstant, Partitioning.MONTHS);
+    assertThat(list, contains(
+        Instant.parse("2017-01-31T00:00:00.00Z"),
+        Instant.parse("2017-02-28T00:00:00.00Z"),
+        Instant.parse("2017-03-28T00:00:00.00Z"))
+    );
+  }
+
+  @Test(expected=IllegalArgumentException.class)
+  public void shouldRaiseRangeOfInstantsStartAfterEnd() throws Exception {
+    final Instant startInstant = Instant.parse("2016-12-31T23:00:00.00Z");
+    final Instant endInstant = Instant.parse("2016-01-01T01:00:00.00Z");
+
+    rangeOfInstants(startInstant, endInstant, Partitioning.HOURS);
+  }
+
+  @Test(expected=IllegalArgumentException.class)
```

Comment:
My reasoning not to return just the `startInstant` was that that would go against the fact that `endInstant` is meant to be exclusive. 
Is your concern the fact that this is declared illegal instead of returning an empty list?